sdk/nodejs/config.ts

```diff
@@ -52,6 +52,82 @@ export class Config {
         return getConfig(this.fullKey(key));
     }
 
+    /**
+     * getEnum loads an optional configuration value by its key, or undefined if it doesn't exist. If the value is not
+     * within the array of legal values, an error will be thrown.
+     *
+     * @param key The key to lookup.
+     * @param values The legal enum values.
+     */
+    public getEnum(key: string, values: string[]): string | undefined {
+        const v = getConfig(this.fullKey(key));
+        if (v !== undefined && values.indexOf(v) === -1) {
+            throw new ConfigEnumError(this.fullKey(key), v, values);
+        }
+        return v;
+    }
+
+    /**
+     * getMinMax loads an optional string configuration value by its key, or undefined if it doesn't exist. If the
+     * value's length is less than or greater than the specified number of characters, this function throws.
```

Comment:
this seems super specific.  i feel like it would be better to just have this function, then have a checkLength helper function that enforces this.  `checkLength(config.get(key), min, max)`